### ISSUE TITLE
Loss Function and model.loss.best checkpoint In ESPNet1

### ISSUE THREAD

**Issue by szmmm** [espnet/espnet#2677]
Hi! I have been using this code base https://github.com/creatorscan/espnet/tree/asrtts3-v.0.4.0/egs/librispeech/asrtts/local to build a TTS system based on Tacotron2 and xvectors. I have a few questions:

1. Could someone explain to me when during decoding and synthesis, what is the loss the model looks at during teacher-forcing learning? In other words where is the loss function defined in the code and do I have control over it? It appears that there are alot kinds of losses (BCE, L1, MSE) at the output and I am not entirely sure which one the model uses during training.

2. Is teacher-forcing decoding and synthesis supported in ESPNet1? If so how can I do it?

**Comment by szmmm** [comment]
Please find attached my `path.sh`

**Comment by szmmm** [comment]
Additional Question. In ESPNet1 there are snapshots and one single `model.loss.best`. If I want to run decode and synthesis from a particular epoch snapshot for debugging, can I just use `--model snapshot.ep.xx` for the `run.sh` script? Or is there any other method if this is not correct, because a `snapshot` file seems to contain more information than a `model.loss.best` and I am not sure if I could I directly use the snapshot. Thank you!
